crates/control/src/services/builder/mod.rs

```diff
@@ -93,9 +102,77 @@ where
     }
 }
 
+async fn process_catalog(
+    catalog: Option<sqlx::types::Json<JsonValue>>,
+) -> Result<JsonValue, Error> {
+    let mut catalog = catalog
+        .ok_or_else(|| Error::InvalidCatalog("Missing catalog definition".to_owned()))?
+        .0;
+
+    inject_storage_mapping(&mut catalog)?;
+    encode_resources(&mut catalog)?;
+
+    Ok(catalog.into())
+}
+
+/// Injects valid StorageMappings into the Catalog. We're setting these up upon
+/// signup and this avoids the need for users to include these in every Build's
+/// catalog json individually.
+fn inject_storage_mapping(catalog: &mut JsonValue) -> Result<(), Error> {
+    let c = catalog
+        .as_object_mut()
+        .ok_or_else(|| Error::InvalidCatalog("Catalog must be an object".to_owned()))?;
+
+    // TODO: Once we start to collect Storage Mapping information during signup,
+    // we can inject their real storage mappings here. Until then, this allows
+    // catalogs created by the UI to actually build successfully.
+    let store = serde_json::json!({"provider": "GCS", "bucket": "flow-example"});
+
+    // TODO: How should we setup global resolution of ops/ storage mappings?
```

Comment:
This handling makes sense for the moment.

In the future I think we'll have a global `ops/` storage mapping, along with a mapping for the tenant. Both would be resolved as foreign entities of the build.